openmdao/core/system.py

```diff
@@ -1615,6 +1616,53 @@ def _init_relevance(self, mode):
         if self._use_derivatives:
             desvars = self.get_design_vars(recurse=True, get_sizes=False, use_prom_ivc=False)
             responses = self.get_responses(recurse=True, get_sizes=False, use_prom_ivc=False)
+
+            # If you have response aliases, check for overlapping indices.
+            aliases = {key: value for key, value in responses.items() if value['path']}
+            if len(aliases) > 0:
+
+                used_idx = {}
+                discrete = self._var_allprocs_discrete
+
+                for name in aliases:
+
+                    if name in discrete['input'] or name in discrete['output']:
+                        continue
+
+                    path = responses[name]['path']
+                    if path is None:
+                        path = name
+
+                    if name not in used_idx and path not in used_idx:
+                        size = self._var_allprocs_abs2meta['output'][path]['global_size']
+                        used_idx[path] = np.zeros(size, dtype=int)
```

Comment:
I still think that at some point it would be worth it to investigate using the sparse matrix trick to check the overlapping indices, because the current method could be fairly expensive in terms of memory at least (creating arrays of full global size for every alias and keeping them all simultaneously in a dict) and most likely the sparse method would be considerably faster for large response variables.

Reply:
It is probably worth looking into at some point. I don't know how we would handle partial or full slices though.

Reply:
Fair point.  You'd have to convert the indexers to arrays, which might cost you some performance, but you could still potentially save a lot of memory if the indices are taking a smallish chunk of the full array.